cpp/bindings/constraints.cpp

```diff
@@ -0,0 +1,109 @@
+#include <pybind11/eigen.h>
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/pytypes.h>
+#include <pybind11/stl.h>
+#include <iostream>
+
+#include <string>
+#include <toppra/geometric_path.hpp>
+#include <toppra/constraint.hpp>
+#include <toppra/constraint/joint_torque.hpp>
+#include <toppra/constraint/linear_joint_acceleration.hpp>
+#include <toppra/constraint/linear_joint_velocity.hpp>
+
+namespace py = pybind11;
+
+#ifdef BUILD_WITH_PINOCCHIO
+// Mind that pinocchio headers **must** be included before boost headers.
+#include <toppra/constraint/joint_torque/pinocchio.hpp>
+#ifdef PINOCCHIO_WITH_PYTHON_INTERFACE
+#include <boost/python.hpp>
```

Comment:
I know it works for you, but I wonder if using both boost and pybind is a good idea?

Reply:
I would say there is not issue, but I am not sure.